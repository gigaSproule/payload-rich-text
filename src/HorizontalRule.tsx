import type { Options } from "./types";

export interface HorizontalRuleData {
  type: "horizontalrule";
  version: number;
}

export type Props = {
  data: HorizontalRuleData;
  options?: Options;
};

export const HorizontalRule = ({ data, options }: Props) =>
  options?.horizontalRule ? options.horizontalRule(data) : <hr />;
