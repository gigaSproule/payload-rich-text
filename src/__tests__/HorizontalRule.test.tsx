import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../HorizontalRule";
import { HorizontalRule } from "../HorizontalRule";
import { createHorizontalRuleData } from "./helpers/createHorizontalRuleData";

describe("HorizontalRule", () => {
  it("should render default hr if no options provided", () => {
    const horizontalRuleData: Props["data"] = createHorizontalRuleData();
    const { container } = render(<HorizontalRule data={horizontalRuleData} />);
    expect(container).toMatchSnapshot();
  });

  it("should render default hr if options provided without horizontalRule", () => {
    const horizontalRuleData: Props["data"] = createHorizontalRuleData();
    const { container } = render(
      <HorizontalRule data={horizontalRuleData} options={{}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with horizontalRule", () => {
    const horizontalRuleData: Props["data"] = createHorizontalRuleData();
    const { container } = render(
      <HorizontalRule
        data={horizontalRuleData}
        options={{
          horizontalRule: (horizontalRule) => {
            return <div>Custom HR - {JSON.stringify(horizontalRule)}</div>;
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
