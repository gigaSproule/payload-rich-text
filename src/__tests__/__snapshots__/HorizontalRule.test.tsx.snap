// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalRule should render custom render if options provided with horizontalRule 1`] = `
<div>
  <div>
    Custom HR - 
    {"type":"horizontalrule","version":1}
  </div>
</div>
`;

exports[`HorizontalRule should render default hr if no options provided 1`] = `
<div>
  <hr />
</div>
`;

exports[`HorizontalRule should render default hr if options provided without horizontalRule 1`] = `
<div>
  <hr />
</div>
`;
